### DISCUSSION TITLE
How to fetch token from keyclock using client certificate(SSL)

### DISCUSSION THREAD

**Discussion by Akash-Adkute** [keycloak/keycloak#19349] [Q&A]
Hello,
I was trying mutual authentication (x509) for keyclock by passing client certificate to keyclock server,keyclock server detects the installed client certificate and user can login without giving user credentials.
ref : https://youtu.be/yq1hzNs1JQU
this is working fine,

**Now i'm trying to fetch token from keyclock by passing client ssl certificate but not able to figure it out how to achive it.**

Thanks

**Comment by KosrenDQ** [comment]
Hi, 
2 years later, the internet is still empty with an answer to this question.
Does someone figure it out? Unfortunately, the Keycloak Documentation https://www.keycloak.org/docs/latest/server_admin/#_client-credentials is also not really helpful.

I appreciate any advice on this!

**Comment by KosrenDQ** [comment]
For anyone who is stumbling on this thread, I solved my issues with this example implementation https://github.com/CarolinaFernandez/keycloak-mtls
